From 7099fb7bc9d2823d87ab06a61b97b0f934ae5c17 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sat, 11 Jan 2025 12:06:12 -0500
Subject: [PATCH] Samsung SMSC: When we receive those weird SMSC, actually
 parse it because even in 2025 we still need SMSC for some carriers

---
 .../internal/telephony/MessagingResponse.java       | 13 +++++++++++--
 .../android/internal/telephony/RadioResponse.java   | 13 +++++++++++--
 2 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/MessagingResponse.java b/src/java/com/android/internal/telephony/MessagingResponse.java
index 3ddf676..2e62599 100644
--- a/src/java/com/android/internal/telephony/MessagingResponse.java
+++ b/src/java/com/android/internal/telephony/MessagingResponse.java
@@ -170,8 +170,17 @@ public class MessagingResponse extends IRadioMessagingResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
-        if(smsc.contains("\"") || smsc.contains(","))
-            smsc = "";
+        if(smsc.contains("\"") || smsc.contains(",")) {
+            android.util.Log.e("PHH", "Got weird SMSC: " + smsc);
+            try {
+                String[] a = smsc.split("\"");
+                smsc = a[1];
+            } catch(Throwable t) {
+                android.util.Log.e("PHH", "Failed parsing weird smsc", t);
+                smsc = "";
+            }
+            android.util.Log.e("PHH", "Patched smsc " + smsc);
+        }
         RadioResponse.responseString(HAL_SERVICE_MESSAGING, mRil, responseInfo, smsc);
     }
 
diff --git a/src/java/com/android/internal/telephony/RadioResponse.java b/src/java/com/android/internal/telephony/RadioResponse.java
index 729e959..a322d04 100644
--- a/src/java/com/android/internal/telephony/RadioResponse.java
+++ b/src/java/com/android/internal/telephony/RadioResponse.java
@@ -1267,8 +1267,17 @@ public class RadioResponse extends IRadioResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
-        if(smsc.contains("\"") || smsc.contains(","))
-            smsc = "";
+        if(smsc.contains("\"") || smsc.contains(",")) {
+            android.util.Log.e("PHH", "Got weird SMSC: " + smsc);
+            try {
+                String[] a = smsc.split("\"");
+                smsc = a[1];
+            } catch(Throwable t) {
+                android.util.Log.e("PHH", "Failed parsing weird smsc", t);
+                smsc = "";
+            }
+            android.util.Log.e("PHH", "Patched smsc " + smsc);
+        }
         responseString(responseInfo, smsc);
     }
 
-- 
2.34.1

